### DISCUSSION TITLE
Remapの更新に関する質問とトラックボールを使用したマウスホイールについての質問。

### DISCUSSION THREAD

**Discussion by lranslot** [Yowkees/keyball#432] [Q&A]
お疲れ様です、キーボール４６，６１を使わせていただいています。
現在Remapに更新がかかりキーマップの変更に火がついていろいろ試しているのですが
気になった点がありました。

１，マウスホイール機能をレイヤー３以外のレイヤーでも使えるように移せないでしょうか？
　今レイヤー３にしてトラックボールを触るとマウスホイールとしてつかえていますがレイヤー２で使いたいな、と思ったことが
　原因で何かしらの方法でユーザーの選択で他レイヤーに移すことはできないでしょうか？
　将来的に左右にトラックボールを置くようにするとレイヤー０でのマウスホイール運用も出てくると思うので
　対応できるならしていただいたほうが将来的にも有用かと思いました。

２，キーボール４６は新しいRemapには対応しないのでしょうか？
　自分のファームウエア入れた時期が悪いのかもしれませんが対応していないように見受けられましたので。

３，Remapへのソース登録はしないのでしょうか？
　もしかしたら１，２がこれで解消してしまうかもしれませんがRemapでのユーザービルドが可能になったようです。
　ただ、ソース登録で編集できるように設定しないといけないようになったようです。トラックボールがあるので
　複雑になっている場所だとは思うのですがユーザビルド出来るとトラックボールにいろいろな機能を
　割り当てることができるようになるはずなのでそれだけ可能性は広がる部分だとは思うのですが。

以上３点です。
作者様へ。飛騨高山で６１を購入したときに数字キーが遠いとおっしゃった意味が使っていくうちにわかりました。
トラックボールをもうちょっと上に持ってこないと確かに遠い気がしますね。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

１について、ソースコードを書き換えることで対応可能です。
keymap.c内の
keyball_set_scroll_mode(get_highest_layer(state) == 3);
これの==3を　==2　もしくは　>=2　に変更していただくと、ご希望の動作になると思います。

２について、Keyball46のファームウェアもRemapに登録しているv1.2.0のファームウェアを書き込み直していただければ、最新のRemapが使用可能です。

３についてはRemapのビルド機能について勉強中でして、ソース登録まで少々お待ちください。対応したいと思います。

また、61へのご意見もありがとうございます。ぜひ用途に合ったキー配列を見つけていただければ幸いです。
以上、よろしくお願いいたします。

**Reply by lranslot** [reply]
ご返答ありがとうございます。早速試してみようと思います。４６もアップデートして使ってみようと思います。
４４も気になるんですよね・・・また余裕ができたら購入しようと思います。
